src/cmap/auth/mongo_credentials.ts

```diff
@@ -68,6 +69,8 @@ export interface AuthMechanismProperties extends Document {
   ALLOWED_HOSTS?: string[];
   /** The resource token for OIDC auth in Azure and GCP. */
   TOKEN_RESOURCE?: string;
+  /** A custom AWS credential provider to use. */
```

Comment:
Could we include the code snippet from the highlights in the docs here, or something like that? I think that most likely this will always be used to call fromNodeProviderChain in "userland" so folks can control the configuration, worth demonstrating that.

Reply:
Updated the comments with examples.